core/src/main/java/io/kestra/core/services/AbstractFlowTriggerService.java

```diff
@@ -0,0 +1,158 @@
+package io.kestra.core.services;
+
+import io.kestra.core.models.conditions.types.MultipleCondition;
+import io.kestra.core.models.executions.Execution;
+import io.kestra.core.models.flows.Flow;
+import io.kestra.core.models.triggers.AbstractTrigger;
+import io.kestra.core.models.triggers.multipleflows.MultipleConditionStorageInterface;
+import io.kestra.core.models.triggers.multipleflows.MultipleConditionWindow;
+import io.kestra.core.runners.RunContextFactory;
+import jakarta.inject.Inject;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public abstract class AbstractFlowTriggerService<S extends MultipleConditionStorageInterface> {
```

Comment:
I don't see the point of having a type parameter here, we usually inject the interface and depending on the runner and implementation is choosed.

Reply:
That's something I forgot to remove as before the save & delete methods weren't in the interface so I had to have access to the impl directly to be able to do such action and that was preventing the need to cast